src/sparkx/Filter.py

```diff
@@ -323,49 +370,46 @@ def spacetime_cut(particle_list, dim, cut_value_tuple):
     list of lists
         Filtered list of lists containing particle objects for each event
     """
-    if not isinstance(cut_value_tuple, tuple) or len(cut_value_tuple) != 2:
-        raise TypeError('Input value must be a tuple of length two')
-    elif any(val is not None and not isinstance(val, (int, float)) for val in cut_value_tuple):
-        raise ValueError('Non-numeric value found in cut_value_tuple')
-    elif cut_value_tuple[0] is None and cut_value_tuple[1] is None:
-        raise ValueError('At least one cut limit must be a number')
-    elif dim == "t" and cut_value_tuple[0] < 0:
-        raise ValueError('Time boundary must be positive or zero.')
+    if not isinstance(cut_value_tuple, tuple):
+        raise TypeError('Input value must be a tuple containing either ' +
+                        'positive numbers or None of length two')
+
+    __ensure_tuple_is_valid_else_raise_error(cut_value_tuple, allow_none=True)
+
     if dim not in ("x", "y", "z", "t"):
         raise ValueError('Only "t, x, y and z are possible dimensions.')
 
     if cut_value_tuple[0] is None:
-        if (dim != "t"):
-            lower_cut = float('-inf')
-        else:
-            lower_cut = 0.0
```

Comment:
You change the allowed value for the lower cut in `t`, but I think that's fine.

Reply:
I have done so because in SMASH we actually have a print out to the Oscar file from the IC that comes with negative times. If a user wants to have a look at times prior to the first collision, it would not be possible with a cut having a lower boundary of 0.0